### ISSUE TITLE
os.path.exists.__name__ is different on Windows since #101196

### ISSUE THREAD

**Issue by digitalresistor** [python/cpython#115012]
# Bug report

### Bug description:

On macOS/Linux and Windows (before Python 3.12):

```python
>>> import os
>>> os.path.exists.__name__
'exists'
```

Since Python 3.12 on Windows:

```python
>>> import os
>>> os.path.exists.__name__
'_path_exists'
```

This was introduced in https://github.com/python/cpython/issues/101196 where optimized calls were made available on Windows.

Now, I will be honest that relying on the `__name__` for `os.path.exists` is not great, but unfortunately I found this when running a test suite where we were relying on the `__name__` being `exists`, which has been the case since at least Python 2.6 when I first took over maintenance of the project.

### CPython versions tested on:

3.8, 3.9, 3.10, 3.11, 3.12

### Operating systems tested on:

Linux, macOS, Windows

**Comment by zooba** [comment]
What was the test checking for?

I don't think we can easily change the name, and many `os.path` functions names will vary based on platform and API availability (e.g. `abspath`).

Arguably it's more useful for the `__name__` to reflect the actual implementation being used than to be hiding it.

**Comment by digitalresistor** [comment]
@zooba checking that `os.path` could be imported and that it had an attribute named `exists.__name__`. It was used in the Waitress test suite to validate that the package loading works correctly, and instead of using something within the package itself to attempt loading `os.path` was used instead.

It was added before my time.

I don't think it was a good test, but I changed it from hardcoding `exists` to `os.path.exists.__name__` and the tests now pass again, so it's no big deal, but it worked for many many years and changing the `__name__` and it not matching the function name felt weird. That being said, if you want it to reflect the actual implementation used, I would argue that it should have information on where it is coming from, `_path_exists` IMHO is not a great name either.
